src/api/ledger/transactions.js

```diff
@@ -29,6 +29,7 @@ function transactionFilter(address, filters, tx) {
     return false;
   }
   if (filters.counterparty && tx.address !== filters.counterparty
+      && tx.specification.destination
```

Comment:
To be defensive, it might be good to also check that `tx.specification` exists in case someone adds a transaction type that doesn't have a specification in the future.

Also, we should probably add support for trustline transactions here.

And for transaction types that do not have a counterparty, this should return false because if there is no counterparty, then no transactions can match the specified counterparty.